base/serialize.jl

```diff
@@ -674,6 +678,11 @@ function deserialize(s::SerializationState, t::DataType)
     end
     if nf == 0
         return ccall(:jl_new_struct, Any, (Any,Any...), t)
+    elseif t.pointerfree && ccall(:jl_datatype_haspadding, Cint, (Any,), t)==0
+        x = ccall(:jl_new_struct_uninit, Any, (Any,), t)
+        a_ = pointer_to_array(convert(Ptr{UInt8},pointer_from_objref(x)), sizeof(t))
```

Comment:
i've thought they should be `unsafe_read!` and implemented consistently instead of the current ad-hoc implementations